frontend/src/employee-frontend/components/child-information/ChildDiscussions.tsx

```diff
@@ -0,0 +1,338 @@
+        <CreationModal
+          childId={childId}
+          onAfterSubmitReload={reloadDiscussionData}
+          onClose={() => setCreationModalOpen(false)}
+        />
+      )}
+      {renderResult(childDiscussionData, (discussions) => (
+        <ChildDiscussionSummary
+          discussions={discussions}
+          reloadDiscussionData={reloadDiscussionData}
+        />
+      ))}
+    </div>
+  )
+})
+
+const CreationModal = React.memo(function CreationModal({
+  childId,
+  onAfterSubmitReload,
+  onClose
+}: {
+  childId: UUID
+  onAfterSubmitReload: () => Promise<Result<unknown>> | null
+  onClose: () => void
+}) {
+  const { i18n, lang } = useTranslation()
+
+  const form = useForm(
+    discussionForm,
+    () => ({
+      offeredDate: null,
+      heldDate: null,
+      counselingDate: null
+    }),
+    i18n.validationErrors
+  )
+
+  const offeredDateState = useFormField(form, 'offeredDate')
+  const heldDateState = useFormField(form, 'heldDate')
+  const counselingDateState = useFormField(form, 'counselingDate')
+
+  const submit = async () => await createChildDiscussion(childId, form.value())
+
+  const handleOnSuccess = async () => {
+    if (onAfterSubmitReload) {
+      await onAfterSubmitReload()
+    }
+    onClose()
+  }
+
+  return (
+    <AsyncFormModal
+      title={i18n.childInformation.childDiscussion.addNew}
+      resolveAction={submit}
+      onSuccess={handleOnSuccess}
+      resolveLabel={i18n.common.confirm}
+      rejectAction={onClose}
+      rejectLabel={i18n.common.cancel}
+      resolveDisabled={!form.isValid()}
+    >
+      <FixedSpaceRow alignItems="center" justifyContent="center">
+        <Label>{i18n.childInformation.childDiscussion.offered}</Label>
+        <DatePickerF
+          bind={offeredDateState}
+          locale={lang}
+          data-qa="add-discussion-offered-date-input"
+        />
+      </FixedSpaceRow>
+      <Gap />
+      <FixedSpaceRow alignItems="center" justifyContent="center">
+        <Label>{i18n.childInformation.childDiscussion.held}</Label>
+        <DatePickerF
+          bind={heldDateState}
+          locale={lang}
+          data-qa="add-discussion-held-date-input"
+        />
+      </FixedSpaceRow>
+      <Gap />
+      <FixedSpaceRow alignItems="center" justifyContent="center">
+        <Label>{i18n.childInformation.childDiscussion.counseling}</Label>
+        <DatePickerF
+          bind={counselingDateState}
+          locale={lang}
+          data-qa="add-discussion-counseling-date-input"
+        />
+      </FixedSpaceRow>
+    </AsyncFormModal>
+  )
+})
+
+const ChildDiscussionSummary = React.memo(function ChildDiscussionSummary({
+  discussions,
+  reloadDiscussionData
+}: {
+  discussions: ChildDiscussionWithPermittedActions[]
+  reloadDiscussionData: () => void
+}) {
+  return (
+    <>
+      {discussions.map((discussion, index) => (
+        <Fragment key={discussion.data.id}>
+          <ChildDiscussionRow
```

Comment:
This doesn't look optimal. Maybe it should be a table like the other sections 🤔 

<img width="1133" alt="Screenshot 2023-07-26 at 8 02 14" src="https://github.com/espoon-voltti/evaka/assets/70927/e903470f-fbd3-411e-9e79-619162e0e7a4">

Reply:
Good point, now I refactored it to be a table.